include/RAJA/pattern/forall.hpp

```diff
@@ -589,6 +589,136 @@ RAJA_INLINE camp::resources::EventProxy<Res> CallForallIcount::operator()(T cons
 
 }  // namespace detail
 
+//
+// Experimental support for dynamic policy selection
+//
+namespace expt
+{
+
+
+
+template<typename T, typename t_pol, typename SEGMENT, typename BODY>
+struct IsHostResource {
+  static const bool value = false;
+
+#if defined(RAJA_DEVICE_ACTIVE)
+  static resources::EventProxy<resources::Resource>
+  invoke_forall(RAJA::resources::Resource r, SEGMENT const &seg, BODY const &body)
+  {
+#if defined(RAJA_CUDA_ACTIVE)
+    RAJA::forall<t_pol>(r.get<RAJA::resources::Cuda>(), seg, body);
```

Comment:
Should this be something more like this?
```suggestion
    RAJA::forall<t_pol>(r.get<T>(), seg, body);
```